### DISCUSSION TITLE
Create a Sensitive Environment Variable on a project with REST API

### DISCUSSION THREAD

**Discussion by BadWolfDev** [vercel/community#7032] [Feature Question]
### Question

Is there any way to add a sensitive env variable to a project using the api? I'm trying to generate keys in a way that they will never be read by a human.

**Comment by versecafe** [comment]
You'd likely need to rebuild to load in the new env vars when it comes to a lot of statically generated content, would using the [Vercel Edge Config](https://vercel.com/storage/edge-config) work better since it's designed for runtime changes for the more programatic usage?

**Reply by amyegan** [reply]
@BadWolfDev If you're talking about tokens that change somewhat frequently, then Edge Config is a more appropriate option for sure. 

For longer lived values you should be able to use the API to [create one or more environment variables](https://vercel.com/docs/rest-api/endpoints/projects#create-one-or-more-environment-variables).
